lib/ansible/modules/network/iosxr/iosxr_config.py

```diff
@@ -260,7 +261,8 @@ def run(module, result):
 
             result['updates'] = commands
 
-        load_config(module, commands, result)
+        load_config(module, commands, not module.check_mode)
+        result['changed'] = not module.check_mode
```

Comment:
Shouldn't the `changed` flag be based on the return.  If `diff` is empy `changed=False` else `changed=True`?